packages/x-components/src/x-installer/x-installer/x-installer.ts

```diff
@@ -276,13 +276,15 @@ export class XInstaller {
   ): Vue | undefined {
     if (this.options.app !== undefined) {
       const vue = this.getVue();
-      this.snippetConfig = vue.observable(snippetConfig);
+      const normalisedSnippetConfig = (this.snippetConfig = vue.observable(
+        this.normaliseSnippetConfig(snippetConfig)
+      ));
```

Comment:
Just to understand it, why do we need to define `normalisedSnippetConfig`? Instead of:
```
this.snippetConfig = vue.observable(this.normaliseSnippetConfig(snippetConfig));
```

Reply:
Good question! I did that because of testing code:

- One of the things I did previously was cloning the client snippet config. I did it because it felt natural for me that as we are going to modify an object provided by an external user, it felt safer to clone it. But I discarded this approach
- `provide` functions have bound Vue `this` context, so it wasn't accessible. I now realised that even if we don't need that normalised variable because we mutate rather than clone, I would change the `provide` syntax to use the object syntax.